vllm/engine/arg_utils.py

```diff
@@ -98,6 +98,49 @@ def union_dict_and_str(val: str) -> Optional[Union[str, dict[str, str]]]:
     return optional_type(json.loads)(val)
 
 
+def parse_limit_mm_per_prompt(val: str):
+    """Parse limit-mm-per-prompt with support for configurable options."""
+    import json
+
+    from vllm.config.multimodal import (AudioDummyOptions, BaseDummyOptions,
+                                        ImageDummyOptions, VideoDummyOptions)
+
+    try:
+        parsed = json.loads(val)
+    except json.JSONDecodeError as e:
+        raise ValueError(
+            f"Invalid JSON format for --limit-mm-per-prompt: {e}") from e
+
+    if not isinstance(parsed, dict):
+        raise ValueError("--limit-mm-per-prompt must be a JSON object")
+
+    result = {}
+    for modality, options in parsed.items():
+        # Normalize int to dict first
+        if isinstance(options, int):
+            options = {"count": options}
+        elif not isinstance(options, dict):
+            raise ValueError(
+                f"Invalid options type for {modality}: {type(options)}. "
+                f"Must be int or dict.")
+
+        # Single path: create ModalityDummyOptions from dict
+        try:
+            if modality == "video":
+                result[modality] = VideoDummyOptions(**options)
+            elif modality == "image":
+                result[modality] = ImageDummyOptions(**options)
+            elif modality == "audio":
+                result[modality] = AudioDummyOptions(**options)
+            else:
+                # Unknown modality - use BaseDummyOptions for OOT models
+                result[modality] = BaseDummyOptions(**options)
+        except TypeError as e:
+            raise ValueError(f"Invalid options for {modality}: {e}") from e
```

Comment:
```suggestion
            raise ValueError(f"Invalid options for {modality}") from e
```

No need to print the exception, it's already in the stack trace